dotcom-rendering/src/lib/useUserPreferredEdition.ts

```diff
@@ -0,0 +1,71 @@
+import { getCookie, removeCookie, setCookie } from '@guardian/libs';
+import { useEffect, useState } from 'react';
+import { mutate } from 'swr';
+import {
+	type EditionId as Edition,
+	getEditionFromPageId,
+	isNetworkFront,
+} from './edition';
+
+const getBannerValueFromQueryParams = () => {
+	const queryParams = new URLSearchParams(window.location.search);
+	return queryParams.get('editionSwitcherBanner');
+};
+
+const addOrRemoveCookie = () => {
+	const value = getBannerValueFromQueryParams();
+	if (value === 'hide') {
+		setCookie({
+			name: 'gu_hide_edition_switcher_banner',
```

Comment:
Do we need to make T&C aware of this new cookie? I believe they maintain a list of all cookies.

Reply:
Good shout, I'll send T&C a message